src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -503,6 +535,129 @@ private static void createBranches(List<LfBus> lfBuses, LfNetwork lfNetwork, LfT
+        if (parameters.isAreaInterchangeControl()) {
+            loadingContext.areaBusMap
+                    .entrySet()
+                    .stream()
+                    .filter(e -> checkBoundariesComponent(e.getKey(), numCC, numSC))
+                    .forEach(e -> {
+                        Area area = e.getKey();
+                        Set<LfBus> lfBuses = e.getValue();
+                        Set<LfArea.Boundary> boundaries = loadingContext.areaBoundaries.getOrDefault(area, new HashSet<>());
+                        LfArea lfArea = LfAreaImpl.create(area, lfBuses, boundaries, network, parameters);
+                        network.addArea(lfArea);
+                        postProcessors.forEach(pp -> pp.onAreaAdded(area, lfArea));
+                    });
+        }
+    }
+
+    private static boolean checkBoundariesComponent(Area area, int numCC, int numSC) {
+        boolean allBoundariesInSameComponent = true;
+        for (AreaBoundary boundary : area.getAreaBoundaryStream().toList()) {
+            Bus bus;
+            if (boundary.getTerminal().isPresent()) {
+                Terminal terminal = boundary.getTerminal().get();
+                bus = terminal.getBusBreakerView().getConnectableBus();
+            } else if (boundary.getBoundary().isPresent()) {
+                DanglingLine danglingLine = boundary.getBoundary().get().getDanglingLine();
+                bus = danglingLine.getTerminal().getBusBreakerView().getConnectableBus();
+            } else {
+                continue;
+            }
+            if (bus.getConnectedComponent() != null && bus.getSynchronousComponent() != null && (bus.getConnectedComponent().getNum() != numCC || bus.getSynchronousComponent().getNum() != numSC)) {
+                LOGGER.warn("Area {} does not have all its boundary buses in the same connected component or synchronous component. The area will not be considered for area interchange control", area.getId());
+                allBoundariesInSameComponent = false;
+                break;
+            }
+        }
+        return allBoundariesInSameComponent;
+    }
+
+    /**
+     * If we perform Area Interchange Control, we need all buses to have an Area or to be boundary buses
+     * A boundary bus is connected to one or multiple buses that are all in areas different from each other
+     * If a bus is not in any area, and is not a boundary bus, the area interchange control could not be performed and an exception will be thrown
+     * However, it is possible that the flow through this bus is not considered for the interchange flow of some of the areas it is connected to.
+     * In this case, the slack injection of this bus will be shared between those areas to ensure a correct balancing
+     */
+
+    private static void checkBusesWithoutArea(LfNetwork network) {
```

Comment:
Delete this function and the one below as it is now dead code

Reply:
fixed in dae8119